molgenis-data-index/src/main/java/org/molgenis/data/index/IndexedRepositoryDecorator.java

```diff
@@ -135,4 +158,89 @@ private boolean querySupported(Query<Entity> q)
 		return !containsAnyOperator(q, unsupportedOperators) && !containsComputedAttribute(q, getEntityType())
 				&& !containsNestedQueryRuleField(q);
 	}
+
+	private long indexCount(Query<Entity> q)
+	{
+		try
+		{
+			return searchService.count(getEntityType(), q);
+		}
+		catch (UnknownIndexException e)
+		{
+			waitForIndexToBeStable();
+
+			// try again
+			try
+			{
+				return searchService.count(getEntityType(), q);
+			}
+			catch (UnknownIndexException e1)
+			{
+				throw translateUnknownIndexException();
+			}
+		}
+	}
+
+	private Object indexSearchOne(Query<Entity> q)
+	{
+		try
+		{
+			return searchService.searchOne(getEntityType(), q);
+		}
+		catch (UnknownIndexException e)
+		{
+			waitForIndexToBeStable();
+
+			// try again
+			try
+			{
+				return searchService.searchOne(getEntityType(), q);
+			}
+			catch (UnknownIndexException e1)
+			{
+				throw translateUnknownIndexException();
+			}
+		}
+	}
+
+	private Stream<Object> indexSearch(Query<Entity> q)
+	{
+		try
+		{
+			return searchService.search(getEntityType(), q);
+		}
+		catch (UnknownIndexException e)
+		{
+			waitForIndexToBeStable();
+
+			// try again
+			try
+			{
+				return searchService.search(getEntityType(), q);
+			}
+			catch (UnknownIndexException e1)
+			{
+				throw translateUnknownIndexException();
+			}
+		}
+	}
+
+	private void waitForIndexToBeStable()
+	{
+		try
+		{
+			indexJobScheduler.waitForIndexToBeStableIncludingReferences(getEntityType());
```

Comment:
This one used to be used in tests only, if we're going to use in production I think it's best to add a timeout. What do you think?

Reply:
Its up to `waitForIndexToBeStableIncludingReferences` to interrupt after a certain amount of time. I propose to keep it like this and evaluate at the end of this sprint if and which changes are desired.